src/Integration.Vsix/CFamily/ProcessRunner.cs

```diff
@@ -116,68 +120,33 @@ public bool Execute(ProcessRunnerArguments runnerArgs)
                     else
                     {
                         LogMessage(CFamilyStrings.MSG_ExecutionCancelled);
-                        return false;
+                        return;
                     }
                 }
 
-                var result = WaitForProcessToFinish(process, runnerArgs);
-                return result;
-            }
-        }
-
-        private bool WaitForProcessToFinish(Process process, ProcessRunnerArguments runnerArgs)
-        {
-            process.BeginErrorReadLine();
-            process.BeginOutputReadLine();
+                process.BeginErrorReadLine();
 
-            // Warning: do not log the raw command line args as they
-            // may contain sensitive data
-            LogDebug(CFamilyStrings.MSG_ExecutingFile,
-                runnerArgs.ExeName,
-                runnerArgs.AsLogText(),
-                runnerArgs.WorkingDirectory,
-                runnerArgs.TimeoutInMilliseconds,
-                process.Id);
-
-            var succeeded = process.WaitForExit(runnerArgs.TimeoutInMilliseconds);
-
-            if (succeeded)
-            {
-                process.WaitForExit(); // Give any asynchronous events the chance to complete
-            }
-
-            // false means we asked the process to stop but it didn't.
-            // true: we might still have timed out, but the process ended when we asked it to
-            if (succeeded)
-            {
-                LogDebug(CFamilyStrings.MSG_ExecutionExitCode, process.ExitCode);
-                ExitCode = process.ExitCode;
-
-                if (process.ExitCode != 0 && !runnerArgs.CancellationToken.IsCancellationRequested)
-                {
-                    LogError(CFamilyStrings.ERROR_ProcessRunner_Failed, process.ExitCode);
-                }
-            }
-            else
-            {
-                ExitCode = ErrorCode;
+                // Warning: do not log the raw command line args as they
+                // may contain sensitive data
+                LogDebug(CFamilyStrings.MSG_ExecutingFile,
+                    runnerArgs.ExeName,
+                    runnerArgs.AsLogText(),
+                    runnerArgs.WorkingDirectory,
+                    process.Id);
 
                 try
                 {
-                    process.Kill();
-                    LogWarning(CFamilyStrings.WARN_ExecutionTimedOutKilled, runnerArgs.TimeoutInMilliseconds,
-                        runnerArgs.ExeName);
+                    runnerArgs.HandleInputStream(process.StandardInput);
+                    runnerArgs.HandleOutputStream(process.StandardOutput);
+                    process.WaitForExit();
+                    ExitCode = process.ExitCode;
                 }
-                catch
+                catch (Exception) when (isRunningProcessCancelled)
```

Comment:
We shouldn't be catching critical exceptions.

Reply:
It's interesting that the previous code used `catch` without an exception type so it would catch non-CLS compliant exception. However, that was probably just a mistake.